### PR TITLE
Create  fence-painting.md

### LINKED ISSUE #770: [Feature Request] <ADD DP ALGO>fence painting problem
I want to implement the fence painting problem algorithm to calculate the number of ways to paint the fence with different colors while ensuring no two adjacent posts are painted the same color.

I would like to add the following C++ algorithm that uses dynamic programming to solve the fence painting problem:

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by nishant4500**
### Description
add one algo on dp 

Fixes #770 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/dynamic-programming/fence-painting.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+// Driver program to test the function
+int main() {
+    int n = 3; // Number of fence posts
+    int k = 2; // Number of colors
+
+    printf("Number of ways to paint the fence: %d\n", countWays(n, k));
+
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/fence-painting.md:78]: please write end of your code example like ```` ``` ````

**Comment by nishant4500** [comment]
can you please @ajay-dhangar merged

**Comment by ajay-dhangar** [comment]
> can you please @ajay-dhangar merged

you need to close your code example with ```` ``` ```` and push I'll merge this

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by nishant4500** [comment]
done @ajay-dhangar !!!

**Pull request merged**